### PR TITLE
feat: APPS-2653 Enable ES Location Listing page

### PR DESCRIPTION
**Pull request by farosFreed**
Connected to [APPS-2653](https://jira.library.ucla.edu/browse/APPS-2653)

**Page/Pages Created/updated:** /visit/locations/index.vue from #2653

**Notes:**

In progress, currently missing filters, suspect correct data is not being returned from getListingFilters?

**Time Report:**

This took me 4 hours (so far) to build this.

**Checklist:**

-   [ X] I added github label for semantic versioning
-   [ ] I double checked it looks like the designs
-   [ ] I completed any required mobile breakpoint styling
-   [ ] I completed any required hover state styling
-   [ ] I included a working spec file
-   [ ] I added notes above about how long it took to build this component
-   [ ] UX has reviewed this PR
-   [ ] I assigned this PR to someone to review

[APPS-2653]: https://uclalibrary.atlassian.net/browse/APPS-2653?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by farosFreed** [comment]
Screenshot 2024-05-02 at 3 24 36 PM

Search Aggs seems to return an empty array of buckets

**Review by jendiamond — CHANGES_REQUESTED** [review]
Should this line:
`const { $graphql, $dataApi } = useNuxtApp()`
be
`const { $graphql, $elasticsearchplugin, $dataApi } = useNuxtApp()`

**Comment by farosFreed** [comment]
@jendiamond I am importing elasticsearch on line 43 when its needed for the indexing operation only, so I don't think I need to import it at the top of the file as well unless we want to standardize this pattern of import.

It is slightly more efficient not to import the library when we aren't going to use it, I believe

File: pages/visit/locations/index.vue [hunk 1]
```diff
@@ -76,74 +76,68 @@ const searchGenericQuery = ref({
...
 })
-
-/* TODO: Refactor when search functionality is ready */
-/*
-  async fetch() {
-  // this.uclaLibraries = []
-  this.hits = []
+// ELASTICSEARCH
+async function searchES() {
+  hits.value = []
```
Comment (pghorpade) [inline pages/visit/locations/index.vue:81]: there is some unwanted code, so replace this function with 
async function searchES() {
  if (
    (route.query.q && route.query.q !== '') ||
    (route.query.filters &&
      queryFilterHasValues(
        route.query.filters,
        config.locationsList.filters
      ))
  ) {
    console.log('Search ES HITS query,', route.query.q)
    const queryText = route.query.q || '*'
    const results = await $dataApi.keywordSearchWithFilters(
      queryText,
      config.locationsList.searchFields,
      "sectionHandle:location OR sectionHandle:affiliateLibrary'",
      (route.query.filters &&
        JSON.parse(route.query.filters)) ||
      {},
      config.locationsList.sortField,
      config.locationsList.orderBy,
      config.locationsList.resultFields,
      []
    )
    if (results && results.hits && results.hits.total.value > 0) {
      console.log('Search ES HITS,', results.hits.hits)
      hits.value = results.hits.hits
      noResultsFound.value = false
    } else {
      noResultsFound.value = true
      hits.value = []
    }
  } else {
    // console.log('data.value', data.value)
    hits.value = []
    noResultsFound.value = false
  }
}

**Review by pghorpade — CHANGES_REQUESTED** [review]

File: pages/visit/locations/index.vue (hunk 1 above)
Comment (farosFreed) [reply pages/visit/locations/index.vue]: just made these changes ✅

**Review by pghorpade — DISMISSED** [review]
👍

**Comment by farosFreed** [comment]
Per slack conversation with Parinita, I've replaced the 2 if checks in the indexing statement with 1 if statement.

`if(data.value.entry.affiliateLibraries && data.value.entry.affiliateLibraries.length > 0 && process.server )`

**Review by jendiamond — APPROVED** [review]

**Pull request merged**